modules/mod_google_transcribe/google_glue.cpp

```diff
@@ -45,13 +47,18 @@ class GStreamer {
 		RecognitionConfig* config = streaming_config->mutable_config();
 
     streaming_config->set_interim_results(interim);
-    if (switch_true(switch_channel_get_variable(channel, "GOOGLE_SPEECH_SINGLE_UTTERANCE"))) {
-      switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(m_session), SWITCH_LOG_DEBUG, "single_utterance\n");
+    if (single_utterence == 1) {
+      switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(m_session), SWITCH_LOG_DEBUG, "enable_single_utterance\n");
       streaming_config->set_single_utterance(true);
     }
 
 		config->set_language_code(lang);
-  	config->set_sample_rate_hertz(8000);
+    switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(m_session), SWITCH_LOG_DEBUG, "transcribe language %s \n", lang);
+    
+    int sample_rate = atoi(switch_channel_get_variable(channel, "read_rate"));
+  	config->set_sample_rate_hertz(sample_rate);
+    switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(m_session), SWITCH_LOG_DEBUG, "sample rate %d \n", sample_rate);
```

Comment:
is `sample_rate` set internally by FS? or do we have to set this before the call to this function?

Reply:
it is done by fs itself